From 3fc92ffc8d6f3e8dce329172dec8926f4b674cba Mon Sep 17 00:00:00 2001
Message-ID: <3fc92ffc8d6f3e8dce329172dec8926f4b674cba.1736059918.git.divya@subvertising.org>
In-Reply-To: <cover.1736059918.git.divya@subvertising.org>
References: <cover.1736059918.git.divya@subvertising.org>
From: Divya Ranjan <divya@subvertising.org>
Date: Sat, 4 Jan 2025 20:38:49 +0000
Subject: [PATCH 05/44] gnu: Add rust-ioctl-rs-0.1.

* gnu/packages/crates-io.scm (rust-ioctl-rs-0.1): New variable.

Change-Id: Ie5f8a2e6d8da7c192367452a74e535289f7a99bb
---
 gnu/packages/crates-io.scm | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/gnu/packages/crates-io.scm b/gnu/packages/crates-io.scm
index 11c36cc7dd..7394fb5072 100644
--- a/gnu/packages/crates-io.scm
+++ b/gnu/packages/crates-io.scm
@@ -47,6 +47,7 @@
 ;;; Copyright © 2024 Jordan Moore <lockbox@struct.foo>
 ;;; Copyright © 2024 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;; Copyright © 2024 Murilo <murilo@disroot.org>
+;;; Copyright © 2024 Divya Ranjan Pattanaik <divya@subvertising.org>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -34816,6 +34817,25 @@ (define-public rust-io-close-0.3
 @code{BufWriter}.")
     (license (list license:expat license:asl2.0))))
 
+(define-public rust-ioctl-rs-0.1
+  (package
+    (name "rust-ioctl-rs")
+    (version "0.1.6")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (crate-uri "ioctl-rs" version))
+       (file-name (string-append name "-" version ".tar.gz"))
+       (sha256
+        (base32 "0zdrgqxblrwm4ym8pwrr7a4dwjzxrvr1k0qjx6rk1vjwi480b5zp"))))
+    (build-system cargo-build-system)
+    (arguments
+     `(#:cargo-inputs (("rust-libc" ,rust-libc-0.2))))
+    (home-page "https://github.com/dcuddeback/ioctl-rs")
+    (synopsis "Rust bindings for system ioctls")
+    (description "This package provides Rust bindings for system ioctls.")
+    (license license:expat)))
+
 (define-public rust-io-extras-0.18
   (package
     (name "rust-io-extras")
-- 
2.47.1

